subprojects/docs/src/docs/userguide/reference/ci-systems/github-actions.adoc

```diff
@@ -0,0 +1,194 @@
+[[build_github_actions]]
+= Executing Gradle builds on GitHub Actions
+
+TIP: Top engineering teams using GitHub Actions have been able to reduce CI build time by up to 90% by using the Gradle Build Cache. https://gradle.com/training/build-cache-deep-dive/?bid=guides-execute-travisci[Register here] for our Build Cache training session to learn how your team can achieve similar results.
+
+Building Gradle projects doesn't stop with the developer's machine.
+https://en.wikipedia.org/wiki/Continuous_integration[Continuous Integration] (CI) has been a long-established practice for running a build for every single change committed to version control to tighten the feedback loop.
+
+In this guide, we'll discuss how to configure link:https://github.com/features/actions/[GitHub Actions] for a Gradle project hosted on GitHub.
+
+== Introduction
+
+GitHub Actions is a free, cloud-based CI solution provider built directly into GitHub making it an excellent choice for open source projects hosted on GitHub.
+You can build any project as long as it is hosted on GitHub.
+GitHub Actions can be configured to make your build outputs and test results available for download.
+
+== What you'll need
+
+* A text editor
+* A command prompt
+* The Java Development Kit (JDK), version 1.8 or higher
+* A local Gradle installation, to initialize a new Gradle project
+* A GitHub account
+
+== Setup a Gradle project on GitHub
+
+If you have an existing Gradle project hosted on GitHub, then you can skip this step and move directly to <<configure-github-actions,the next section>>.
```

Comment:
That internal reference to `configure-github-actions` is broken. In order for it to not break our automated check, the section must have an explicite ID, like `[[configure-github-actions]]` on top of the title it represents.